src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_IncrementalAnalyzer.cs

```diff
@@ -92,7 +96,10 @@ private async Task AnalyzeDocumentForKindAsync(TextDocument document, AnalysisKi
                     foreach (var (analyzer, state) in nonCachedAnalyzersAndStates)
                     {
                         var computedData = await ComputeDocumentAnalysisDataAsync(executor, analyzer, state, logTelemetry, cancellationToken).ConfigureAwait(false);
-                        PersistAndRaiseDiagnosticsIfNeeded(computedData, analyzer, state);
+
+                        PersistAndAddDiagnosticsArgsIfNeeded(ref argsBuilder.AsRef(), computedData, analyzer, state);
+
+                        AnalyzerService.RaiseDiagnosticsUpdated(argsBuilder.ToImmutableAndClear());
```

Comment:
Please add a comment here stating that we intentionally raise DiagnosticsUpdates separately for each analyzer so that we don't wait for all analyzer execution to complete before refreshing diagnostics for each completed analyzer.

Reply:
➡️ Added comment